imgserve.go

```diff
@@ -27,12 +30,60 @@ func main() {
 
 //initRoutes initializes all routes
 func initRoutes(directory string) {
-	http.Handle("/", http.FileServer(http.Dir(directory)))
+	http.Handle("/", logInfo(http.FileServer(http.Dir(directory)), directory))
 	http.HandleFunc("/info/", func(w http.ResponseWriter, r *http.Request) {
 		getImageInfo(w, r, directory)
 	})
 }
 
+func logInfo(next http.Handler, directory string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		timeStart := time.Now()
+		clientID := "clientID: " + r.RemoteAddr
+		log.Println(clientID + ", start: " + timeStart.Format(time.RFC3339))
+		next.ServeHTTP(w, r)
+
+		timeEnd := time.Now()
+		log.Println(clientID + ", end: " + timeEnd.Format(time.RFC3339))
+
+		imageName := strings.TrimPrefix(r.URL.Path, "/")
+		if imageName == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("name can not be empty"))
+			return
+		}
+
+		averageSpeed, err := countAverageSpeed(directory, imageName, timeEnd.Unix()-timeStart.Unix())
+		if err != nil {
```

Comment:
Let's just log error here without returning error to the client. Point is: we transferred a potentially large image, we should not undo that work.